### PR TITLE
Use created & updated instead BeforeInsert & BeforeUpdate

### PR DESCRIPTION
**Pull request by lunny**
Also update xorm to the latest version.

### CODE REVIEW

File: models/models.go [hunk 1]
```diff
@@ -239,6 +239,7 @@ func NewTestEngine(x *xorm.Engine) (err error) {
 
 	x.SetMapper(core.GonicMapper{})
 	x.SetLogger(log.XORMLogger)
+	x.ShowSQL(true)
```
Comment (ethantkoenig) [inline models/models.go]: What does this line do?

**Review by ethantkoenig — APPROVED** [review]
Looks good, just one quick question

File: models/models.go (hunk 1 above)
Comment (lunny) [reply models/models.go]: For debug to show the SQL

Comment (ethantkoenig) [reply models/models.go]: So should we remove it?

Comment (lafriks) [reply models/models.go]: This should probably only be set to true only when running Gitea in development mode

**Comment by lafriks** [comment]
Had no rights to push to your repository. But correct checksums for govendor are:
```
+++ b/vendor/vendor.json
@@ -456,7 +456,7 @@
                        "revisionTime": "2017-05-19T03:21:30Z"
                },
                {
-                       "checksumSHA1": "XJ7SsJiRegdDEiIUFjsMNVnNP+w=",
+                       "checksumSHA1": "HMavuxvDhKOwmbbFnYt9hfT6jE0=",
                        "path": "github.com/go-xorm/core",
                        "revision": "da1adaf7a28ca792961721a34e6e04945200c890",
                        "revisionTime": "2017-09-09T08:56:53Z"
@@ -468,7 +468,7 @@
                        "revisionTime": "2016-08-11T02:11:45Z"
                },
                {
-                       "checksumSHA1": "6R9OhB9axG1EoeTYxcIkdscvPNU=",
+                       "checksumSHA1": "SoZDnkd5NdwE0g4MY1nTAl8ZtPo=",
                        "path": "github.com/go-xorm/xorm",
                        "revision": "c29485f954ed6495861c0098f9a1642a467299c8",
                        "revisionTime": "2017-09-10T12:58:18Z"
```

**Review by ethantkoenig — APPROVED** [review]

**Comment by ethantkoenig** [comment]
LGTM

**Comment by bkcsoft** [comment]
LGTM

**Comment by lafriks** [comment]
I think these files are missing changes:
`gpg_key.go`
`issue_stopwatch.go`
`issue_tracked_time.go`

**Comment by lunny** [comment]
@lafriks those `BeforeInsert` are some complicated and I will send another PRs to do that.

**Pull request merged**